src/dlr.cc

```diff
@@ -181,38 +181,29 @@ extern "C" int GetDLROutputByName(DLRModelHandle* handle, const char* name, void
   API_END();
 }
 
-std::vector<std::string> MakePathVec(const char* model_path) {
-  /* Logic to handle Windows drive letter */
-  std::string model_path_string{model_path};
-  std::string special_prefix{""};
-  if (model_path_string.length() >= 2 && model_path_string[1] == ':' &&
-      std::isalpha(model_path_string[0], std::locale("C"))) {
-    // Handle drive letter
-    special_prefix = model_path_string.substr(0, 2);
-    model_path_string = model_path_string.substr(2);
-  }
-
-  std::vector<std::string> path_vec = dmlc::Split(model_path_string, ':');
-  path_vec[0] = special_prefix + path_vec[0];
+std::vector<std::string> MakePathVec(std::string model_path) {
+  std::vector<std::string> path_vec = dmlc::Split(model_path, ':');
+  for (int i = 0; i < path_vec.size(); i++) path_vec[i] = FixWindowsDriveLetter(path_vec[i]);
   return path_vec;
 }
 
 DLRModelPtr CreateDLRModelPtr(const char* model_path, DLContext& ctx) {
   std::vector<std::string> path_vec = MakePathVec(model_path);
-  DLRBackend backend = dlr::GetBackend(path_vec);
+  std::vector<std::string> files = FindFiles(path_vec);
+  DLRBackend backend = dlr::GetBackend(files);
   if (backend == DLRBackend::kTVM) {
-    return std::make_shared<TVMModel>(path_vec, ctx);
+    return std::make_shared<TVMModel>(files, ctx);
   } else if (backend == DLRBackend::kRELAYVM) {
-    return std::make_shared<RelayVMModel>(path_vec, ctx);
+    return std::make_shared<RelayVMModel>(files, ctx);
   } else if (backend == DLRBackend::kTREELITE) {
-    return std::make_shared<TreeliteModel>(path_vec, ctx);
+    return std::make_shared<TreeliteModel>(files, ctx);
 #ifdef DLR_HEXAGON
   } else if (backend == DLRBackend::kHEXAGON) {
-    const std::string model_path_string(model_path);
-    return std::make_shared<HexagonModel>(model_path_string, ctx, 1 /*debug_level*/);
+    return std::make_shared<HexagonModel>(files, ctx, 1 /*debug_level*/);
 #endif  // DLR_HEXAGON
   } else {
-    throw dmlc::Error("Unsupported backend!");
+    LOG(ERROR) << "Unable to determine backend from path: '" << model_path << "'.";
+    return nullptr;  // unreachable
```

Comment:
LOG(ERROR) isnt fatal (that's LOG(FATAL)) so this is actually reachable. Can you instead throw dmlc::Error? dmlc::Error will be caught higher up and cause the C API function to return -1.

Reply:
done